src/api/handlers/PetFindManyHandler.ts

```diff
@@ -1,21 +1,22 @@
-import { Handler, ProxyResult } from "aws-lambda";
+import { APIGatewayEvent, Handler, ProxyResult } from "aws-lambda";
+import { DatabaseServerlessHandler } from "../core/DatabaseServerlessHandler";
 import { ProxyResultBuilder } from "../core/ProxyResultBuilder";
 import { PetService } from "../services/PetService";
 
-class PetFindManyHandler {
-    private petService: PetService;
+class PetFindManyHandler extends DatabaseServerlessHandler<APIGatewayEvent> {
+    private petService: PetService | undefined;
 
-    constructor() {
-        this.petService = new PetService();
+    initializeDependencies(): void {
+        this.petService = new PetService(this.connection);
     }
 
-    public execute(): ProxyResult {
-        const response = this.petService.findMany();
+    public async onHandleEvent(): Promise<ProxyResult> {        
+        const response = await this.petService.findMany();
         return new ProxyResultBuilder().status(200).body(response).build();
     }
 }
 
 export const handler: Handler = async (event, context, callback) => {
-    const response = new PetFindManyHandler().execute();
+    const response = new PetFindManyHandler().execute(event);
```

Comment:
@Aristevao here we are creating a new `PetFindManyHandler` object and **right now after the creation we call the execute method**. The `execute` method is an `async` function, **so we need to fix this call in all handler files** to make the `handler` function awaits for the response.